cfme/v2v/conversion_hosts.py

```diff
@@ -0,0 +1,239 @@
+    class results(View):  # noqa
+        msg = Text('.//h3[contains(@class,"blank-slate-pf-main-action")]')
+
+        @property
+        def is_displayed(self):
+            return self.msg.is_displayed
+
+
+@attr.s
+class ConversionHost(BaseEntity):
+    """Class representing conversion Host
+        Args:
+            target_provider: RHV or OSP
+            provider_name: rhv43 or osp13-ims
+            cluster: Default
+            hostname: host to be configured
+            conv_host_key: /etc/pki/ovirt-engine/keys/engine_id_rsa on RHV-M.
+            transformation_method: vddk or ssh
+            vddk_library_path: url for vddk
+            vmware_ssh_key: /var/lib/vdsm/.ssh/id_rsa on rhv host
+            osp_ca_cert : CA cert
+
+    """
+    target_provider = attr.ib()
+    provider_name = attr.ib()
+    cluster = attr.ib()
+    hostname = attr.ib()
+    conv_host_key = attr.ib()
+    transformation_method = attr.ib()
+    vddk_library_path = attr.ib(default=None)
+    vmware_ssh_key = attr.ib(default=None)
+    osp_cert_switch = attr.ib(default=None)
+    osp_ca_cert = attr.ib(default=None)
+
+    @property
+    def is_host_configured(self):
+        """Wait for conversion host to configure and return success/failure state"""
+        view = navigate_to(self.parent, "All")
+        try:
+            wait_for(
+                lambda: not view.conversion_host_progress.in_progress(self.hostname),
+                message="Conversion Host configuration is in progress",
+                delay=5,
+                num_sec=300
+            )
+        except TimedOutError:
+            self.logger.warning("Timed out waiting for {} to configure".format(self.hostname))
+        return view.conversion_host_progress.is_host_configured(self.hostname)
+
+
+@attr.s
+class ConversionHostCollection(BaseCollection):
+    """Collection object for conversion host object"""
+
+    ENTITY = ConversionHost
+
+    def create(self,
+               target_provider,
+               provider_name,
+               cluster,
+               hostname,
+               conv_host_key,
+               transformation_method,
+               vddk_library_path=None,
+               vmware_ssh_key=None,
+               osp_cert_switch=None,
+               osp_ca_cert=None):
+        """Configure new conversions host in UI
+        Args:
+            target_provider: RHV or OSP
+            provider_name: rhv43 or osp13-ims
+            cluster: Default
+            hostname: Hostname to be configured
+            conv_host_key: /etc/pki/ovirt-engine/keys/engine_id_rsa on RHV-M.
+            transformation_method: vddk or ssh
+            vddk_library_path: url for vddk
+            vmware_ssh_key: /var/lib/vdsm/.ssh/id_rsa on rhv host
+            osp_cert_switch: Yes or No
+            osp_ca_cert: Tls Cert
+
+        """
+        view = navigate_to(self, "Configure")
+        provider_type = ("Red Hat Virtualization" if target_provider.one_of(RHEVMProvider)
```

Comment:
This string is being repeated now in a few places, and should be defined as a Provider class attribute for each provider class.

Looks like EC2 and Azure already have 'discover_name` class attribute, though discovery is unique to those providers now, it may be a good attribute to use to set these pretty names on the various provider types.

Reply:
Can I do this in separate PR ?